src/http-requests.ts

```diff
@@ -40,19 +40,34 @@ function constructHostURL(host: string): string {
   }
 }
 
+function cloneAndParseHeaders(headers: HeadersInit): Record<string, string> {
+  if (Array.isArray(headers)) {
+    return headers.reduce((acc, headerPair) => {
+      acc[headerPair[0]] = headerPair[1]
+      return acc
+    }, {} as Record<string, string>)
+  } else if ('has' in headers) {
+    const clonedHeaders: Record<string, string> = {}
+    ;(headers as Headers).forEach((value, key) => (clonedHeaders[key] = value))
+    return clonedHeaders
+  } else {
+    return Object.assign({}, headers)
+  }
+}
```

Comment:
Sorry I did not ask this sooner.

Could you add two tests, one with headers as a Map and one as a nested array?

Example for nested array:

```ts
  const client = new MeiliSearch({
        ...config,
        apiKey: '',
        requestConfig: {
          headers: [['Authorization', `Bearer ${key}`]], 
        },
      })
```

example with Map()
```ts
      const headers = new Map()
      headers.set('Authorization', `Bearer ${key}`)

      const client = new MeiliSearch({
        ...config,
        apiKey: '',
        requestConfig: {
          // @ts-ignore
          headers,
        },
      })
```

As you can see I added a `ts-ignore` on the Map test.  It seems that [`HeadersInit`](https://microsoft.github.io/PowerBI-JavaScript/modules/_node_modules_typedoc_node_modules_typescript_lib_lib_dom_d_.html#headersinit) does not accept `Map`. 

If you think it is a common use case we can keep it and keep the ignore comment, but I'm very curious if you have any example of a lib or a use-case where Map is used.

Reply:
sure 

it's not a map it accepts Headers object just use new Headers() instead will add tests

Reply:
added tests